jax/custom_derivatives.py

```diff
@@ -951,3 +955,162 @@ def merge(l1, l2):
 
 def abstractify(x):
   return core.raise_to_shaped(core.get_aval(x))
+
+
+### Custom transposition
+
+def linear_call(fun: Callable, fun_transpose: Callable, residual_args,
+                linear_args):
+  """Call a function, indicating a custom implementation for its transpose.
```

Comment:
```suggestion
  """Call a linear function, indicating a custom implementation for its transpose.
```

Would it make sense to mention 'linear' up-front here?